cirq/sim/density_matrix_utils.py

```diff
@@ -257,3 +257,16 @@ def _validate_indices(num_qubits: int, indices: List[int]) -> None:
     if any(index >= num_qubits for index in indices):
         raise IndexError('Out of range indices, must be less than number of '
                          'qubits but was {}'.format(indices))
+
+
+def von_neumann_entropy(density_matrix: np.ndarray) -> float:
+    """Calculates von Neumann entropy of density matrix in bits.
+    Args:
+        density_matrix: The density matrix.
+    Returns:
+        The calculated von Neumann entropy.
+    """
+    eigenvalues = np.linalg.eigvals(density_matrix)
+    if 0 in eigenvalues:
+        return 0
```

Comment:
This is like saying that a die roll has no entropy because it can never roll a 7. You're not multiplying by zero when you see zero, you're adding zero when you see zero.

```
    def h(p: float):
        p = abs(p)
        if p == 0:
            return 0
        return -p * math.log2(p)

    return sum(h(p) for p in np.linalg.eigvals(density_matrix))
```

Reply:
Makes sense. I haven't been able to find anything online that mentions this "nuance" on the definition of entropy. Everything I've found just gives the formula. How is one supposed to know what to do when an eigenvalue is 0? Any pointers? Thanks.

CCing @bryano cause i discussed this with him too.

Reply:
Like Craig said in the sync, this is because the limit as p -> 0 of p log p is 0. One way of seeing this is to write p log p as log p / (1 / p) and apply L'Hospital's rule.

At an operational level, this also has to be the case because events of zero probability shouldn't affect the information content of the distribution.